en/ui-system/components/editor/videoplayer.md

```diff
@@ -0,0 +1,175 @@
+
+**Note**: On iOS platform, due to the platform limitations, the CLICKED event can't be fired when VideoPlayer is in fullscreen mode. If you want to let the Video played in fullscreen and also fire the CLICKED event properly, you should use a Widget component to hack the VideoPlayer's size.
+
+For more information, please refer to the [VideoPlayer Events](../../../api/en/classes/VideoPlayer.html#events) or [21.video-player](https://github.com/cocos-creator/test-cases-3d/tree/v3.0/assets/cases/ui/21.video-player) of the test-cases-3d samples bundled with Creator.
+
+## Detailed Explanation
+
+The supported video types is **mp4** format.
+
+### Add a callback via script
+
+#### Method one
+
+This method uses the same API that editor uses to add an event callback on Button component. You need to construct a `cc.Component.EventHandler` object first, and then set the corresponding `target`, `component`, `handler` and `customEventData` parameters.
+
+```ts
+import { _decorator, Component, VideoPlayer } from 'cc';
+const { ccclass, property } = _decorator;
+
+@ccclass('cc.MyComponent')
+export class MyComponent extends Component {
+    @property(VideoPlayer)
+    videoPlayer = null;
+
+    start () {
+        const eventHandler = new Component.EventHandler();
+        eventHandler.target = newTarget;
+        eventHandler.component = "cc.MyComponent";
+        eventHandler.handler = "callback";
+        eventHandler.customEventData = "foobar";
+        this.videoplayer.videoPlayerEvent.push(eventHandler);
+    }
+
+    // the order of parameters should not change
+    callback: function(videoplayer, eventType, customEventData) {
+        // videoplayer is a VideoPlayer component instance
+        // eventType is typed as cc.VideoPlayer.EventType 
+        // customEventData is "foobar"
+    }
+}
+```
+
+#### Method two
+
+Add event callback with `videoplayer.node.on('ready-to-play', ...)`
+
+```ts
+// Suppose we add event handling callbacks in the onLoad method of a component and perform event handling in the callback function:
+import { _decorator, Component, find, VideoPlayer } from 'cc';
+const { ccclass, property } = _decorator;
+
+@ccclass('VideoPlayerCtrl')
+export class VideoPlayerCtrl extends Component {
+    @property(VideoPlayer)
+    videoPlayer = null;
+
+    start () {
+        this.videoplayer.node.on('ready-to-play', this.callback, this);
+    }
+
+    callback (event) {
+        // event is EventCustom, you can use event.detail to get VideoPlayer component
+        let videoplayer = event.detail;
+        // do whatever you want with videoplayer
+        // you can't pass customEventData in this way
+    }
+}
+```
+
+Likewise, you can also register `meta-loaded`, `clicked`, `playing` events, and the parameters of the callback function for these events are consistent with the `ready-to-play` parameters.
+
+About VideoPlayer events, please refer to [VideoPlayer API](../../../api/en/classes/VideoPlayer.html) for details.
```

Comment:
About VideoPlayer events, please refer to [VideoPlayer API](../../../api/en/classes/VideoPlayer.html) for details.
-->
Please refer to the [VideoPlayer API](../../../api/en/classes/VideoPlayer.html) documentation for details on __VideoPlayer events__.